test/built-ins/Array/prototype/every/resizable-buffer-grow-mid-iteration.js

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.every
+description: >
+  Array.p.every behaves correctly when receiver is backed by resizable
+  buffer that is grown mid-iteration
+includes: [compareArray.js, resizableArrayBufferUtils.js]
+features: [resizable-arraybuffer]
+---*/
+
+const ArrayEveryHelper = (ta, values, rab, resizeAfter, resizeTo) => {
+  return Array.prototype.every.call(
+    ta,
+    (n) => CollectValuesAndResize(n, values, rab, resizeAfter, resizeTo));
+};
+
+// Orig. array: [0, 2, 4, 6]
+//              [0, 2, 4, 6] << fixedLength
+//                    [4, 6] << fixedLengthWithOffset
+//              [0, 2, 4, 6, ...] << lengthTracking
+//                    [4, 6, ...] << lengthTrackingWithOffset
+let values;
+let rab;
+let resizeAfter;
+let resizeTo;
+for (let ctor of ctors) {
+  rab = CreateRabForTest(ctor);
+  const fixedLength = new ctor(rab, 0, 4);
+  values = [];
+  resizeAfter = 2;
+  resizeTo = 5 * ctor.BYTES_PER_ELEMENT;
```

Comment:
Probably better to scope these values inside the loops, to prevent accidental contamination. It doesn't look like they need to be shared between the loops.

```suggestion
for (let ctor of ctors) {
  const rab = CreateRabForTest(ctor);
  const fixedLength = new ctor(rab, 0, 4);
  const values = [];
  const resizeAfter = 2;
  const resizeTo = 3 * ctor.BYTES_PER_ELEMENT;
```